### PR TITLE
feat(startup checks): Ensure enough memory is available before starting

### PR DESCRIPTION
**Pull request by vikaspotluri123**
Refs #47

### CODE REVIEW

**Comment by vikaspotluri123** [comment]
Should I also add in update checks, or should that be done as a separate PR?

I had started this before winter holidays, but hadn't gotten around to testing it. Just added tests 😄

File: lib/commands/doctor/index.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ class DoctorCommand extends Command {
             return Promise.reject(e);
         }
 
-        return this.ui.listr(checks, {ui: this.ui, system: this.system}).then(() => {
+        return this.ui.listr(checks, {argv: argv, ui: this.ui, system: this.system}).then(() => {
```
Comment (acburdine) [inline lib/commands/doctor/index.js]: I'd say we should remove this bit (and the related arguments) for now. Definitely 👍 with being able to skip it - but we can add it back in when we refactor the `ghost doctor` command (which will happen hopefully sometime this week)

**Comment by vikaspotluri123** [comment]
@acburdine 👍 done

**Comment by acburdine** [comment]
@vikaspotluri123 looks like there's some conflicts with the latest changes in master (I refactored the doctor checks & split each check out into a separate file - would you mind doing the same with this change?

**Comment by vikaspotluri123** [comment]
Yeah, I figured it would be better to wait on rebasing until you said something 😜 I'll do it either 2/4 night (CST) or 2/5 morning 👍

**Comment by acburdine** [comment]
No rush - thanks!

**Comment by vikaspotluri123** [comment]
Should I also implement an update mem-check for 200(?) MB?

**Comment by acburdine** [comment]
Hmm....I'd say maybe just increase the memory limit to 150mb and run it on both start and update? That should be enough I think.

**Comment by vikaspotluri123** [comment]
Not sure if I explicitly said it, but this is ready 👍

File: lib/commands/doctor/index.js (hunk 1 above)
Comment (vikaspotluri123) [reply lib/commands/doctor/index.js]: With this am I good to re-add the skip?

Comment (acburdine) [reply lib/commands/doctor/index.js]: Sorry for not responding to this until now - I'm good with re-adding the skip. However, that does mean that any command that _calls_ ghost doctor (e.g. `start`, `install`, `update`) will need to be updated to also pull in the required `ghost doctor` args.

Comment (vikaspotluri123) [reply lib/commands/doctor/index.js]: Done! 👍

File: lib/commands/doctor/checks/check-memory.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+const os = require('os');
+const SystemError = require('../../../errors').SystemError;
+
+const MB_IN_BYTES = 1048576;
+const MIN_MEMORY = 150;
+
+function checkMemory(ctx, task) {
+    if (!ctx.argv['check-mem']) {
+        return task.skip();
```
Comment (acburdine) [inline lib/commands/doctor/checks/check-memory.js]: I wonder if it's worth moving this to an `enable` function? That way the check wouldn't show up in the list if it's been intentionally ignored 🤔

Comment (vikaspotluri123) [reply lib/commands/doctor/checks/check-memory.js]: Done!

**Comment by AileenCGN** [comment]
Hey @acburdine is there a reason why this is approved but not merged yet?

**Comment by vikaspotluri123** [comment]
Isn't the option already there?

**Comment by acburdine** [comment]
yeah it's there - what I meant by "push" is advertise it more in the docs & such.

**Pull request merged**